### PR TITLE
[CDF-6715] Update docs for functions

### PR DESCRIPTION
**Pull request by HMEiding**

### CODE REVIEW

File: cognite/experimental/_api/functions.py [hunk 1]
```diff
@@ -42,13 +42,22 @@ def create(
...
         secrets: Optional[Dict] = None,
     ) -> Function:
-        """`Create a new function from source code located in folder. <https://docs.cognite.com/api/playground/#operation/post-api-playground-projects-project-functions>`_
+        """`When creating a function, <https://docs.cognite.com/api/playground/#operation/post-api-playground-projects-project-functions>`_
+        the source code can be specified in one of three ways:\n
+        - Via the `folder` argument, which is the path to the folder where the source code is located. The folder must contain a file named handler.py whithin which a function named handle must be defined.\n
+        - Via the `file_id` argument, which is the ID of a zip-file uploaded to the files API. The zip-file must contain a file named handler.py whithin which a function named handle must be defined.\n
+        - Via the `function_handle` argument, which is a reference to a function object, which must be named `handle`.\n
+
+        The function named `handle` is the entrypoint of the created function. Valid arguments to `handle` are `data`, `client` and `secret`:\n
```
Comment (andeplane) [inline cognite/experimental/_api/functions.py:51]:
```suggestion
        The function named `handle` is the entrypoint of the created function. Valid arguments to `handle` are `data`, `client` and `secrets`:\n
```

File: cognite/experimental/_api/functions.py [hunk 2]
```diff
@@ -42,13 +42,22 @@ def create(
...
+        - If the user calls the function with input data, this is passed through the `data` argument.\n
+        - If the user gives an `api_key` when creating the function, a pre instantiated CogniteClient is passed through the `client` argument.\n
+        - If the user gives one ore more secrets when creating the function, these are passed through the `secrets` argument.\n
```
Comment (andeplane) [inline cognite/experimental/_api/functions.py:54]:
```suggestion
        - If the user gives one ore more secrets when creating the function, these are passed through the `secrets` argument. The API key can be access through `secrets["apikey"]`.\n
```

File: cognite/experimental/_api/functions.py [hunk 3]
```diff
@@ -42,13 +42,22 @@ def create(
         api_key: Optional[str] = None,
         secrets: Optional[Dict] = None,
     ) -> Function:
-        """`Create a new function from source code located in folder. <https://docs.cognite.com/api/playground/#operation/post-api-playground-projects-project-functions>`_
+        """`When creating a function, <https://docs.cognite.com/api/playground/#operation/post-api-playground-projects-project-functions>`_
+        the source code can be specified in one of three ways:\n
+        - Via the `folder` argument, which is the path to the folder where the source code is located. The folder must contain a file named handler.py whithin which a function named handle must be defined.\n
+        - Via the `file_id` argument, which is the ID of a zip-file uploaded to the files API. The zip-file must contain a file named handler.py whithin which a function named handle must be defined.\n
```
Comment (matzhaugen) [inline cognite/experimental/_api/functions.py:48]: whithin -> within

File: cognite/experimental/_api/functions.py [hunk 4]
```diff
@@ -346,11 +355,11 @@ def list(
 
         Args:
             function_id (int, optional): ID of the function on which the calls were made.
+            function_external_id (str, optional): External ID of the function on which the calls were made.
             status (str, optional): Status of the call. Possible values ["Running", "Failed", "Completed", "Timeout"].
             schedule_id (int, optional): Schedule id from which the call belongs (if any).
-            start_time (Union[Dict[str, int], TimestampRange]): Start time of the call. Possible keys are `min` and `max`, with values given as time stamps in ms.
-            end_time (Union[Dict[str, int], TimestampRange]): End time of the call. Possible keys are `min` and `max`, with values given as time stamps in ms.
-            function_external_id (str, optional): External ID of the function on which the calls were made.
+            start_time (Dict[str, int], optional): Start time of the call. Possible keys are `min` and `max`, with values given as time stamps in ms.
```
Comment (matzhaugen) [inline cognite/experimental/_api/functions.py:361]: I suspect these time types can only be `int`, what do you think? https://github.com/cognitedata/context-api/blob/master/src/api/functions/schemas.py#L175-L177

Comment (matzhaugen) [reply cognite/experimental/_api/functions.py:361]: Ah, I understand, this means string keys and int values. In that case, great!

File: cognite/experimental/_api/functions.py (hunk 3 above)
Comment (HMEiding) [reply cognite/experimental/_api/functions.py]: Yup, thanks!

**Pull request merged**